dist/znk.iap.js

```diff
@@ -112,273 +114,651 @@
+                                $ionicLoading.hide();
+                                console.log('purchase: isPurchaseInProgress=' + iapSrv.isPurchaseInProgress);
+                            };
+                            
+                            /////////////////////////////
+                            /////////////////////////////
+                            // Cancelled and Refunded App products Handler
+                            /////////////////////////////
+                            /////////////////////////////
+
+                            iapSrv.appProductsArr.forEach(function (appProduct){
+                                $window.store.when(appProduct.id).cancelled(function(product){
+                                    purchaseCancelled(product);                               
+                                });
+                                $window.store.when(appProduct.id).refunded(function(product){
+                                    console.log('purchase refunded, product:' + product.id);                               
+                                });
+                            });
 
-                    if (!IapSrv._products.length) {
-                        $window.store.refresh();
-                    }
+                            /////////////////////////////
+                            /////////////////////////////
+                            // Updated and Finished App products Handler
+                            /////////////////////////////
+                            /////////////////////////////
+
+                            $window.store.ready(function(){
+                                console.log('-----store is ready-----');
+                                if(!iapStoreTimedOut){
+                                    $timeout.cancel(iapStoreReadyTimeout);
+                                    if (angular.isDefined(iapStoreReadyDfd)){
+                                        iapStoreReadyDfd.resolve();
+                                    }
+                                    
+                                }
+                            }); 
+
+                            iapSrv.appProductsArr.forEach(function (appProduct){
+                                $window.store.when(appProduct.id).updated(function(product){
+                                    console.log('product updated: ' + product.id);
+                                    iapSrv.products[product.id] = product;
+                                });
+                                $window.store.when(appProduct.id).finished(function(product){
+                                    console.log('product finished: ' + product.id);
+                                    $analytics.eventTrack('purchased', { category: 'purchase', label:product.id });
+                                    //hack - for android purposes only
+                                    $analytics.pageTrack('product-purchased/' + product.id);
```

Comment:
It is needed only because our current analytics doesn't work with eventTrack on android. Zion is working on it now. I wanted it to be conistent